Automattic/newspack-newsletters
# Epic/tags support

### All Submissions:






### Changes proposed in this Pull Request:

This PR merges the epic branch we have been working on with the Local (tag based) subscription lists.



### How to test the changes in this Pull Request:

1. Set your ESP as Mailchimp
2. Create a couple of Local lists in Newsletter > Lists
3. Register as a new user to your site
4. Subscribe to a newsletter
5. In Mailchimp dashboard, confirm the user was created with the expected tag
6. Go to My Account > Newsletters
7. Confirm that the page correctly shows the newsletters you have subscribed to
8. Check and uncheck different newsletters and save
9. Confirm that everytime the correct tags are applied to the contact on Mailchimp and the checkboxes are properly updated
10. In the admin (as an admin) switch the ESP to Active Campaign
11. Back to your reader user, go to My Account > Newsletters again
12. All newsletters should be unchecked
13. This user does not exist in the ESP yet, let's see if it will be created on the fly
14. Check one or several newsletters and submit
15. Confirm the user is created in the ESP
16. Play again with checking and unchecking Lists and see if they are correctly updated
17. Check you error log for warnings and errors.

### Other information:
